src/codegen.ts

```diff
@@ -1,39 +1,57 @@
-import { CodegenConfig } from "@graphql-codegen/cli";
+import { writeFileSync } from "fs";
+import path from "path";
 
-const config: CodegenConfig = {
-  schema: "https://api.fragment.dev/schema.graphql",
-  emitLegacyCommonJSImports: false,
-  generates: {
-    "generated/generated.ts": {
-      config: {
-        scalars: {
-          ID: "string",
-          JSON: "Record<string, unknown>",
+import { generate } from "@graphql-codegen/cli";
+import type { Types } from "@graphql-codegen/plugin-helpers";
+import prettier from "prettier";
+
+const GENERATED_FILENAME = "generated/generated.ts";
+
+generate(
+  {
+    schema: "https://api.fragment.dev/schema.graphql",
+    emitLegacyCommonJSImports: false,
+    generates: {
+      [GENERATED_FILENAME]: {
+        documents: "src/queries/**/*.graphql",
+        config: {
+          scalars: {
+            ID: "string",
+            JSON: "Record<string, unknown>",
+          },
+          withHooks: true,
+          defaultScalarType: "string",
+          useImplementingTypes: true,
         },
-        withHooks: true,
-        defaultScalarType: "string",
-        useImplementingTypes: true,
-      },
-      documents: "src/queries/**/*.graphql",
-      hooks: {
-        afterOneFileWrite: [
-          `sed -i '' "s/import gql from 'graphql-tag'/import { gql } from 'graphql-tag'/"`,
-          "prettier --write",
+        plugins: [
+          "typescript",
+          "typescript-operations",
+          "typescript-graphql-request",
+          {
+            add: {
+              content:
+                "/* This file was generated by `yarn gen-clients`. Don't edit it directly */ \n/* eslint-disable @typescript-eslint/no-explicit-any */",
+            },
+          },
         ],
       },
-      plugins: [
-        "typescript",
-        "typescript-operations",
-        "typescript-graphql-request",
-        {
-          add: {
-            content:
-              "/* This file was generated by `yarn gen-clients`. Don't edit it directly */ \n/* eslint-disable @typescript-eslint/no-explicit-any */",
-          },
-        },
-      ],
     },
   },
-};
-
-export default config;
+  false,
+)
+  .then(async ([fileOutput]: Types.FileOutput[]) => {
+    const output = fileOutput.content.replace(
+      /import gql from 'graphql-tag'/g,
+      "import { gql } from 'graphql-tag'",
+    );
+    const prettifiedOutput = await prettier.format(output, {
+      parser: "typescript",
+    });
+    const qualifiedFilename = path.resolve(process.cwd(), fileOutput.filename);
+    writeFileSync(qualifiedFilename, prettifiedOutput, "utf-8");
+    process.exit(0);
+  })
+  .catch((error) => {
+    console.error(error);
+    throw error;
```

Comment:
should we `throw error` or just `system.exit(1)`? `throw error` will result in it being logged a second time, right?

Reply:
Thats right. Good catch!